File: php/src/Models/AppPortalCapability.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

enum AppPortalCapability: string implements \JsonSerializable
{
    case VIEW_BASE = 'ViewBase';
    case VIEW_ENDPOINT_SECRET = 'ViewEndpointSecret';
    case MANAGE_ENDPOINT_SECRET = 'ManageEndpointSecret';
    case MANAGE_TRANSFORMATIONS = 'ManageTransformations';
    case CREATE_ATTEMPTS = 'CreateAttempts';
    case MANAGE_ENDPOINT = 'ManageEndpoint';

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return self::from($data);
    }

    public function jsonSerialize(): mixed
    {
        return $this->value;
    }
}

File: rust/src/models/app_portal_capability.rs
// this file is @generated
use std::fmt;

use serde::{
    Deserialize,
    Serialize,
};

#[derive(
    Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize,
)]
pub enum AppPortalCapability {
    #[default]
    #[serde(rename = "ViewBase")]
    ViewBase,
    #[serde(rename = "ViewEndpointSecret")]
    ViewEndpointSecret,
    #[serde(rename = "ManageEndpointSecret")]
    ManageEndpointSecret,
    #[serde(rename = "ManageTransformations")]
    ManageTransformations,
    #[serde(rename = "CreateAttempts")]
    CreateAttempts,
    #[serde(rename = "ManageEndpoint")]
    ManageEndpoint,
}

impl fmt::Display for AppPortalCapability {
    fn fmt(
        &self,
        f: &mut fmt::Formatter,
    ) -> fmt::Result {
        let value = match self {
            Self::ViewBase => "ViewBase",
            Self::ViewEndpointSecret => "ViewEndpointSecret",
            Self::ManageEndpointSecret => "ManageEndpointSecret",
            Self::ManageTransformations => "ManageTransformations",
            Self::CreateAttempts => "CreateAttempts",
            Self::ManageEndpoint => "ManageEndpoint",
        };
        f.write_str(value)
    }
}
